sinonjs/sinon
# SEO improvements and banner on old documentation

#### Purpose
 Fixes #1923 by adding banner on pages containing docs for older versions of Sinon.
 The banner notifies that the page is old and points to the newest version.
![old-version](https://user-images.githubusercontent.com/618076/46842669-2bb4e580-cdcf-11e8-8148-af953309a043.png)

 It also improves SEO by obliterating all the hits for old versions of Sinon. Instead,
 using rel=canonical, all those search hits now leads to /releases/latest/<page>.


 #### Background
When searching for "Sinon spy" on Google, the most common hits are for version 1.17 and version 2.0.0. This leads to people reporting errors. We have discussed ways of improving the situation, and the solution of having all the doc pages point to the latest version of the same page using rel=canonical seemed like the most pragmatic way of getting better search hits while still keeping the old docs around for reference.
![skjermdump fra 2018-10-11 21-53-30](https://user-images.githubusercontent.com/618076/46842663-18097f00-cdcf-11e8-9737-4dcf67bd88a8.png)

 #### Solution  - optional
The lack of a clear way of seperating programming logic from the view logic
meant some bits have a slightly hacky feel to them. You might suggest other
ways of doing so :-)

 #### How to verify - mandatory
1. Check out this branch
2. `bundle exec jekyll serve`
3. Browse the [/releases page](http://localhost:4000/releases/)
4. Observe the banner being there on v6.3.4 but not 6.3.5
5. Type http://localhost:4000/releases/latest/ and see that it shows the docs for 6.3.5
